eng/Subsets.props

```diff
@@ -121,7 +121,8 @@
     <!-- CLR NativeAot only builds in a subset of the matrix -->
     <_NativeAotSupportedOS Condition="'$(TargetOS)' == 'windows' or '$(TargetOS)' == 'linux' or '$(TargetOS)' == 'osx' or '$(TargetOS)' == 'maccatalyst' or '$(TargetOS)' == 'iossimulator' or '$(TargetOS)' == 'ios' or '$(TargetOS)' == 'tvossimulator' or '$(TargetOS)' == 'tvos' or '$(TargetOS)' == 'freebsd'">true</_NativeAotSupportedOS>
     <_NativeAotSupportedArch Condition="'$(TargetArchitecture)' == 'x64' or '$(TargetArchitecture)' == 'arm64' or '$(TargetArchitecture)' == 'arm' or ('$(TargetOS)' == 'windows' and '$(TargetArchitecture)' == 'x86')">true</_NativeAotSupportedArch>
-    <NativeAotSupported Condition="'$(_NativeAotSupportedOS)' == 'true' and $(_NativeAotSupportedArch) == 'true'">true</NativeAotSupported>
+    <NativeAotSupported Condition="'$(_NativeAotSupportedOS)' == 'true' and '$(_NativeAotSupportedArch)' == 'true'">true</NativeAotSupported>
+    <NativeAotCanExecuteOnHost Condition="'$(NativeAotSupported)' == 'true' and ('$(CrossBuild)' != 'true' or '$(TargetOS)' == '$(HostOS)')">true</NativeAotCanExecuteOnHost>
```

Comment:
Why do we need `'$(CrossBuild)' != 'true'` in the condition? I would expect `'$(TargetOS)' == '$(HostOS)'` to be enough.

Reply:
This part of the condition was adapted from main https://github.com/dotnet/runtime/blob/cb71b60b08b3bfd8c0cc17372240931ff2818eff/src/coreclr/tools/aot/crossgen2/crossgen2_publish.csproj#L13-L14

e.g. x64 binary running on arm64 OS is supported (as long as the host and target OS were same).

Reply:
> x64 binary running on arm64 OS is supported (as long as the host and target OS were same).

This is situation should be covered by '$(TargetOS)' != '$(HostOS)' condition. I do not understand why we have `CrossBuild` in the mix.

Reply:
True, I have updated it, lets see in the CI. The matrix is a bit tricky to cover with compile-native.proj in the mix.